### DISCUSSION TITLE
How do we plan on adding patients to groups

### DISCUSSION THREAD

**Discussion by dubdabasoduba** [opensrp/fhircore#298] [Ideas]
**Context:** 
The location mapping sync by [team & location approach](https://github.com/google/android-fhir/issues/263#issuecomment-863203350) has a group to which patients are assigned. We (OpenSRP web) team want to create a module that performs crud functions on these groups. Currently, add the patients to the groups is out of scope for us but, we would like to confirm with everyone here just to make sure we are on the same

**Comment by dubdabasoduba** [comment]
Do we assign patients to groups through the app when the patient is being created? If so do we want to allow OpenSRP web the power to edit these assignments?

**Reply by dubdabasoduba** [reply]
1. Do we assign patients to groups through the app when the patient is being created?  - Will this require us to create the group on the app? or will the groups be synced from the server upon login by the facility worker (This means the groups can be created from OpenSRP FHIR web or any other platform that is linked to HAPI FHIR?
2. If we are to sync the groups from the server, do we want to allow OpenSRP web the power to edit these assignments i.e add and remove patients or familly (groups) from a specific group?

cc: @f-odhiambo

**Reply by dubdabasoduba** [reply]
I think the syncing of groups to after login would work since by this time we can tell the care team the user is assigned to and through the care team we can team the groups this care team is responsible for

**Comment by f-odhiambo** [comment]
@dubdabasoduba Check issue #276. This will resolve this discussion point

**Comment by f-odhiambo** [comment]
There are constraints of using group as not supported in $apply method for CarePlan generation

**Reply by pld** [reply]
I think this is about groups for sync management vs for family representation, @dubdabasoduba is that correct?

**Reply by f-odhiambo** [reply]
I think it still applies but let's wait for Ben's feedback
